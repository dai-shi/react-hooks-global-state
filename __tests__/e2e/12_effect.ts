import puppeteer from 'puppeteer';

jest.setTimeout(15 * 1000);

describe('02_typescript', () => {
  const port = process.env.PORT || '8080';

  it('should work with events', async () => {
    const browser = await puppeteer.launch();
    const page = await browser.newPage();
    await page.goto(`http://localhost:${port}/`);

    await page.click('div:nth-of-type(1) > button:nth-of-type(1)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.click('div:nth-of-type(1) > button:nth-of-type(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.click('div:nth-of-type(2) > button:nth-of-type(1)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await page.click('div:nth-of-type(2) > button:nth-of-type(2)');
    expect(await page.evaluate(() => document.body.innerHTML)).toMatchSnapshot();

    await browser.close();
  });
});
