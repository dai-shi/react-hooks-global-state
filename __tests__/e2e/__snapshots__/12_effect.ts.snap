// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`02_typescript should work with events 1`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>Bonus: 40</span></div><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>Bonus: 40</span></div></div>
  

"
`;

exports[`02_typescript should work with events 2`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>Bonus: 60</span></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>Bonus: 60</span></div></div>
  

"
`;

exports[`02_typescript should work with events 3`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>Bonus: 80</span></div><div><span>Count: 1</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>Bonus: 80</span></div></div>
  

"
`;

exports[`02_typescript should work with events 4`] = `
"
    <div id=\\"app\\"><h1>Counter</h1><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>Bonus: 100</span></div><div><span>Count: 0</span><button type=\\"button\\">+1</button><button type=\\"button\\">-1</button><span>Bonus: 100</span></div></div>
  

"
`;
